tools/rum-models-generator/Sources/RUMModelsGeneratorCore/Input/JSON/JSONSchemaToJSONTypeTransformer.swift

```diff
@@ -7,7 +7,7 @@
 import Foundation
 
 /// Transforms ambiguous `JSONSchema` into type-safe `JSONObject` schema.
-internal class JSONSchemaToJSONTypeTransformer {
+internal class JSONSchemaToJSONTypeTransformer: TypeTransformer<(name: String, schema: JSONSchema)> {
```

Comment:
Now that we inherit from `TypeTransformer`, we should implement [this method](https://github.com/DataDog/dd-sdk-ios/blob/develop/tools/rum-models-generator/Sources/RUMModelsGeneratorCore/Output/Preparing/TypeTransformer.swift#L29) as well.

Since the `context` is only used internally, it might be easier to just add a property `let context: TransformationContext<(name: String, schema: JSONSchema)>`. And if this property is `private`, we could get rid of the `precondition`! WDYT?

Reply:
We can definitely keep it 👍  I was referencing to the fact that by using a `private` context, we manage the cohesion within the class: the context cannot be altered externally.